File: t-tasks/src/main/java/com/teo/ttasks/ui/activities/edit_task/EditTaskView.java
package com.teo.ttasks.ui.activities.edit_task;

import android.app.DatePickerDialog.OnDateSetListener;
import android.app.TimePickerDialog.OnTimeSetListener;
import android.view.View;

import com.teo.ttasks.data.model.TTask;
import com.teo.ttasks.data.model.TTaskList;
import com.teo.ttasks.ui.base.MvpView;

import java.util.List;

public interface EditTaskView extends MvpView, OnDateSetListener, OnTimeSetListener {

    void onTaskLoaded(TTask task);

    void onTaskListsLoaded(List<TTaskList> taskLists, int selectedPosition);

    void onTaskLoadError();

    void onTaskSaved(TTask task);

    void onTaskSaveError();

    boolean onDueDateLongClicked(View v);

    boolean onDueTimeLongClicked(View v);

    void onTitleChanged(CharSequence title, int start, int before, int count);

    void onNotesChanged(CharSequence notes, int start, int before, int count);
}

File: t-tasks/src/main/java/com/teo/ttasks/ui/activities/edit_task/EditTaskView.kt
package com.teo.ttasks.ui.activities.edit_task

import android.app.DatePickerDialog.OnDateSetListener
import android.app.TimePickerDialog.OnTimeSetListener
import android.view.View
import com.teo.ttasks.data.model.Task
import com.teo.ttasks.data.model.TaskList
import com.teo.ttasks.ui.base.MvpView

internal interface EditTaskView : MvpView, OnDateSetListener {

    fun onTaskLoaded(task: Task)

    fun onTaskListsLoaded(taskLists: List<TaskList>, selectedPosition: Int)

    fun onTaskLoadError()

    fun onTaskSaved()

    fun onTaskSaveError()

    fun onDueTimeLongClicked(v: View): Boolean

    fun onTitleChanged(title: CharSequence, start: Int, before: Int, count: Int)

    fun onNotesChanged(notes: CharSequence, start: Int, before: Int, count: Int)
}
